internal/matcher/controller.go

```diff
@@ -142,7 +142,7 @@ func (mc *Controller) filter(ctx context.Context, interested []*claircore.IndexR
 		if err != nil {
 			return nil, err
 		}
-		filtered[record.Package.ID] = match
+		filtered[record.Package.ID] = append(filtered[record.Package.ID], match...)
```

Comment:
For context:
* We get the relevant indexing records by leveraging the matcher's `Filter()` method
* We pass those records to the query function to get back a map of potential vulns (`map[record.Package.ID][]vuln`).
* When checking if `Vulnerable()` we iterate through the records and check if it (the record) is vulnerable to any of it's potential vulns.
* The matched vulns are save in the `filtered` map we're referring to
* Index records are create as one per package-repository combination

So we have a situation where previously the `record.Package.ID` was specific enough because there was only one `record` with that `record.Package.ID`  *because* we were Querying on the `RepoName`. Now there are mutiple `records` with the same `record.Package.ID` but different `record.Repo`s. Hence, the last `record` with a random `Repo` would overwrite any potential legitimate vulnerabilities if the matches weren't appended

Reply:
ah ok, so this is called prior to `Vulnerable()` and we need all potential CPEs, as `Vulnerable()` will do the CPE matching from now on. If this is correct, feel free to resolve